src/transformers/models/bartpho/tokenization_bartpho.py

```diff
@@ -0,0 +1,302 @@
+# coding=utf-8
+# Copyright 2021 VinAI Research and the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+""" Tokenization classes for BARTpho-syllable model."""
+
+
+import os
+from shutil import copyfile
+from typing import Any, Dict, List, Optional, Tuple
+
+import sentencepiece as spm
+
+from ...tokenization_utils import AddedToken, PreTrainedTokenizer
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+SPIECE_UNDERLINE = "▁"
+
+VOCAB_FILES_NAMES = {"vocab_file": "sentencepiece.bpe.model", "reduced_vocab_file": "dict.txt"}
```

Comment:
The vocab file name for  [bartpho-word](https://huggingface.co/vinai/bartpho-word/tree/main) is `vocab.txt`, it should also be changed to `dict.txt` otherwise it' 'll fail to load, since the file names need to match with specified file names here.

Reply:
No, it will not. BartphoTokenizer is used for `bartpho-syllable` only. 

`barpho-word` employs PhobertTokenizer, as shown in its `config.json` file and our paper: https://arxiv.org/pdf/2109.09701.pdf

The difference between `bartpho-syllable` and `bartpho-word` is that they take different types of input texts: syllable level for `bartpho-syllable` (e.g. 7-syllable written text “chúng tôi là những nghiên cứu viên”) vs. word level for `barpho-word` (e.g. 4-word text “chúng_tôi là những nghiên_cứu_viên”). So they use different tokenizers.